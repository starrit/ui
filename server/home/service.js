import React from 'react';
import { renderToString } from 'react-dom/server';
import template from '../template';

module.exports = class CareService {

  /**
   * Gets Promise of PreRendered Care Page.
   * Returns null if current etag would match etag passed in.
   * Error is thrown if data for desired region/locality/type is not present.
   * @param existingEtag passed in etag for comparison.
   * @param type type of care centers desired.
   * @param region region of care centers desired.
   * @param locality locality of care centers desired.
   * @param page current page requested.
   * @param path path of current request.
   * @returns {Promise.<T>}
   */
  getHomePage(existingEtag, type, region, locality, page, path) {
    // GET Care data from API.
    React.createElement()
    const appString = renderToString(<h1>Hello</h1>)
    const out = {
      body : template({
        body: appString,
        title: 'StarrIT L.L.C. Providing a complete set of consulting services to meet your business needs. ',
        initialState: JSON.stringify({}),
        cssBundle: '', //ENV['main.css'],
        jsBundle: 'dist/main.js'
      })};

    return out;
  }

  /**
   * Validates data returned from API to ensure that data for given locality exists.
   * @param data data returned from API.
   * @param care_type type of desired care centers.
   * @param region region of desired care centers.
   * @param locality of desired care centers.
   */
  validateCareData(data, care_type, region, locality)  {

    // ensure that 'meta' section of response contains at least one item including locality information.
    const meta = data['meta'];
    let locality_info;
    for (var i in meta) {
      if (meta[i]['locality']) {
        locality_info = meta[i]['locality'];
        break;
      }
    }
    if (!locality_info) {
      throw new Error(`Locality info not found for: ${care_type}/${region}/${locality}`)
    }

    // ensure that the type of at least one included section is 'destination_page'
    const included = data['included'];
    let cms_content;
    for (var j in included) {
      if (included[j]['type'] === 'destination_page') {
        cms_content = true;
        break;
      }
    }
    if (!cms_content) {
      throw new Error(`CMS content not found for: ${care_type}/${region}/${locality}`)
    }
  }

  postCareService(type, region, locality) {
    return Promise.resolve();
  }
};
