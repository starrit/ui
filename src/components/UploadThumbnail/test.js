import React from 'react';
import {configure, mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({adapter: new Adapter()});

import {UploadThumbnail} from './';
import * as thumbnailUploadService from 'AppServices/thumbnail-upload.service';
jest.mock('AppServices/thumbnail-upload.service');

describe('UploadThumbnail component', () => {
  beforeAll(() => {
    thumbnailUploadService.uploadThumbnail = (file, url, progressFunction, completeFunction) => {
      setTimeout(() => {
        progressFunction({
          loaded: file.size * file.loaded / 100,
          total: file.size
        });
      });
      setTimeout(() => {
        completeFunction();
      }, 100);
    };
  });

  it('should render componenet', () => {
    const wrapper = shallow(<UploadThumbnail uploadUrl="" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should open file browser', () => {
    const spyOnFocusInput = jest.spyOn(UploadThumbnail.prototype, 'focusInput');
    const wrapper = mount(<UploadThumbnail uploadUrl="" />);
    wrapper.find('.uploadThumbnail').simulate('click');
    expect(spyOnFocusInput).toHaveBeenCalled();
  });

  it('should fail uploading thumbnail', () => {
    const addNotificationMock = jest.fn();
    const wrapper = mount(<UploadThumbnail uploadUrl="" addNotification={addNotificationMock} />);

    wrapper.instance().onImageSelect({
      type: 'plain/text'
    });

    expect(addNotificationMock).toHaveBeenCalledWith({
      text: 'File type not supported',
      type: 'negative'
    });
  });

  it('should call onImageSelect on selecting Image', () => {
    const spyOnImageSelect = jest.spyOn(UploadThumbnail.prototype, 'onImageSelect');
    const wrapper = mount(<UploadThumbnail uploadUrl="" addNotification={jest.fn()} />);
    wrapper.find('input').simulate('change');
    expect(spyOnImageSelect).toHaveBeenCalled();
  });

  it('should upload thumbnail', done => {
    const addNotificationMock = jest.fn();
    const onChangeMock = jest.fn();
    const wrapper = mount(<UploadThumbnail uploadUrl="" onChange={onChangeMock} addNotification={addNotificationMock} />);

    wrapper.instance().onImageSelect(new Blob([''], {type: 'image/png'}));

    setTimeout(() => {
      expect(addNotificationMock).toHaveBeenCalledWith({
        text: 'Thumbnail uploaded successfully',
        type: 'positive'
      });
      expect(onChangeMock).toHaveBeenCalled();
      done();
    }, 1000);
  });
});
