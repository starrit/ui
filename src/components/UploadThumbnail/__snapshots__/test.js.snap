// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadThumbnail component should render componenet 1`] = `
<React.Fragment>
  <Card
    className="thumbnailCard"
  >
    <Thumbnail
      className=""
      default="/assets/images/add_thumbnail.png"
      src={null}
    />
    <input
      className="thumbnailInput"
      name=""
      onChange={[Function]}
      type="file"
    />
  </Card>
  <P
    className="thumbnailText"
  >
    <span
      className="thumbnailCopy pullLeft"
    >
      Thumbnail
    </span>
    <span
      className="pullRight"
    >
      <a
        className="iconWrap uploadThumbnail"
        onClick={[Function]}
      >
        <Icon
          className="thumbnailIcons"
          viewBoxHeight={50}
          viewBoxWidth={50}
        >
          <path
            d="M45,16.4c-0.3-0.4-0.8-0.7-1.3-0.7H33c-0.9,0-1.7,0.7-1.7,1.6c0,0.9,0.7,1.6,1.7,1.6h9.3v26.3H7.8V19h9.3c0.9,0,1.7-0.7,1.7-1.6c0-0.9-0.7-1.6-1.7-1.6H6.5c-0.5,0-1,0.2-1.3,0.7c-0.3,0.2-0.6,0.7-0.6,1.2v29c0,0.5,0.3,1,0.7,1.3c0.3,0.4,0.8,0.7,1.3,0.7h37.2c0.5,0,1-0.2,1.3-0.7c0.4-0.3,0.7-0.8,0.7-1.3v-29C45.7,17.2,45.4,16.7,45,16.4z M17.3,12.2c0.5,0,0.9-0.1,1.3-0.5l4.7-4.1v20.3c0,0.9,0.8,1.6,1.8,1.6c1,0,1.8-0.7,1.8-1.6V7.2l4.7,4.1c0.7,0.6,1.9,0.6,2.5,0c0.7-0.6,0.7-1.6,0-2.3l-7.6-6.7c-0.7-0.6-1.9-0.6-2.5,0l-8,7c-0.7,0.6-0.7,1.6,0,2.3C16.3,12,16.8,12.2,17.3,12.2z"
          />
        </Icon>
      </a>
    </span>
  </P>
</React.Fragment>
`;
